megamek/src/megamek/common/AmmoType.java

```diff
@@ -402,30 +402,43 @@ public boolean equals(Object other) {
         }
         // there a couple of flags that need to be checked
         if (getAmmoType() == T_MML) {
-            if (hasFlag(F_MML_LRM) != ((AmmoType) other).hasFlag(F_MML_LRM)) {
+            if (!matchingMMLFlags((AmmoType) other)) {
                 return false;
             }
         }
         if (getAmmoType() == T_AR10) {
-            if (hasFlag(F_AR10_BARRACUDA) != ((AmmoType) other)
-                    .hasFlag(F_AR10_BARRACUDA)) {
+            if (!matchingAR10Flags((AmmoType) other)) {
                 return false;
             }
-            if (hasFlag(F_AR10_WHITE_SHARK) != ((AmmoType) other)
-                    .hasFlag(F_AR10_WHITE_SHARK)) {
-                return false;
-            }
-            if (hasFlag(F_AR10_KILLER_WHALE) != ((AmmoType) other)
-                    .hasFlag(F_AR10_KILLER_WHALE)) {
+        }
+        return ((getAmmoType() == ((AmmoType) other).getAmmoType()) && (getRackSize() == ((AmmoType) other)
+                .getRackSize()));
+    }
+
+    /**
+     * When comparing <code>AmmoType</code>s, look at the ammoType only.
+     *
+     * @param other the <code>Object</code> to compare to this one.
+     * @return <code>true</code> if the other is an <code>AmmoType</code> object
+     *         of the same <code>ammoType</code> as this object. N.B. different
+     *         munition types are still equal.
+     */
+    public boolean equalsAmmoTypeOnly(Object other) {
+        if (!(other instanceof AmmoType)) {
+            return false;
+        }
+        // there a couple of flags that need to be checked
+        if (getAmmoType() == T_MML) {
+            if (!matchingMMLFlags((AmmoType) other)) {
                 return false;
             }
-            if (hasFlag(F_NUCLEAR) != ((AmmoType) other)
-                    .hasFlag(F_NUCLEAR)) {
+        }
+        if (getAmmoType() == T_AR10) {
+            if (!matchingAR10Flags((AmmoType) other)) {
                 return false;
             }
         }
-        return ((getAmmoType() == ((AmmoType) other).getAmmoType()) && (getRackSize() == ((AmmoType) other)
-                .getRackSize()));
+        return getAmmoType() == ((AmmoType) other).getAmmoType();
     }
```

Comment:
I'm almost certain this will cause issues relating to being able to reuse AC2 ammo for AC20, etc.

Reply:
Actually you might be right I forgot to test that use case. Let me setup a new test campaign. My debug sessions were following the code path of missile racks and not ACs. The Acs might eventually reach the same predicate and use the new equals method which would cause a different bug.

Reply:
You were absolutely correct about the AC ammo. I have made a change in MekHQ so the predicate is based on the AMMO_BY_TYPE set. With this adjustment AC ammo is being compared correctly.